docs/multi-cloud-deployment-guide.md

```diff
@@ -0,0 +1,489 @@
+  name: hello-world
+  namespace: cnpg-system
+spec:
+  ports:
+  - port: 9090
+    protocol: TCP
+    targetPort: 9090
+  selector:
+    app: hello-world
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: hello-world
+  name: hello-world
+  namespace: cnpg-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: hello-world
+  strategy: {}
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: hello-world
+    spec:
+      containers:
+      - args:
+        - plugin
+        - --server-cert=/server/tls.crt
+        - --server-key=/server/tls.key
+        - --client-cert=/client/tls.crt
+        - --server-address=:9090
+        image: pgcosmoscontroller.azurecr.io/cnpg-plugin:v15
+        name: cnpg-i-hello-world
+        ports:
+        - containerPort: 9090
+          protocol: TCP
+        resources: {}
+        volumeMounts:
+        - mountPath: /server
+          name: server
+        - mountPath: /client
+          name: client
+      volumes:
+      - name: server
+        secret:
+          secretName: helloworld-server-tls
+      - name: client
+        secret:
+          secretName: helloworld-client-tls
+EOF
+
+kubectl apply -f sidecar-injector.yaml
+```
+
+4. Install the DocumentDB operator on the hub:
+
+```bash
+VERSION=36  # Replace with your current version
+helm package documentdb-chart --version 0.0.${VERSION}
+helm install documentdb-operator ./documentdb-operator-0.0.${VERSION}.tgz --namespace documentdb-operator-ns 
+```
+
+5. Deploy certificates on each cluster:
+
+```bash
+cat <<EOF > certs.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cnpg-system
+
+---
+
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: helloworld-client
+  namespace: cnpg-system
+spec:
+  commonName: helloworld-client
+  duration: 2160h
+  isCA: false
+  issuerRef:
+    group: cert-manager.io
+    kind: Issuer
+    name: selfsigned-issuer
+  renewBefore: 360h
+  secretName: helloworld-client-tls
+  usages:
+  - client auth
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: helloworld-server
+  namespace: cnpg-system
+spec:
+  commonName: hello-world
+  dnsNames:
+  - hello-world
+  duration: 2160h
+  isCA: false
+  issuerRef:
+    group: cert-manager.io
+    kind: Issuer
+    name: selfsigned-issuer
+  renewBefore: 360h
+  secretName: helloworld-server-tls
+  usages:
+  - server auth
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: selfsigned-issuer
+  namespace: cnpg-system
+spec:
+  selfSigned: {}
+EOF
+
+kazure apply -f certs.yaml
+konprem apply -f certs.yaml
+```
+
+## Deploying DocumentDB Operator to fleet
+
+```bash
+cat <<EOF > documentdb-base.yaml
+apiVersion: placement.kubernetes-fleet.io/v1beta1
+kind: ClusterResourcePlacement
+metadata:
+  name: documentdb-base
+spec:
+  resourceSelectors:
+    - group: ""
+      version: v1
+      kind: Namespace
+      name: documentdb-operator-ns
```

Comment:
Our operator by default gets installed in `documentdb-operator` namespace. Is there way to configure it. Or use the default one?